src/engine/engine.cc

```diff
@@ -35,6 +55,18 @@ inline Engine* CreateEngine() {
   if (type == nullptr) type = "ThreadedEnginePerDevice";
   std::string stype = type;
 
+#ifdef _TEST_FLOAT_EXCEPTION
+  struct sigaction act;
+  memset(&act, '\0', sizeof(act));
+  act.sa_sigaction = &float_num_err_handler;
+  act.sa_flags = SA_SIGINFO;
+  if (sigaction(SIGFPE, &act, NULL) < 0) {
+    fprintf(stderr, "Cannot register signal handler\n");
```

Comment:
Might be also good to log the errno in the message such that we can use it to see why it failed.